sstables/kl/reader_impl.hh

```diff
@@ -295,13 +297,13 @@ private:
                 _deleted = (mask & column_mask::deletion) != column_mask::none;
                 _counter = false;
                 _state = state::CELL;
+                break;
             }
-            break;
```

Comment:
> Why did you change the single "break" into three ones in all three branches of the if? Seems to me just longer and more complicated, isn't it?

It makes it more complicated at the moment, but it's useful later when we remove the switch (and breaks) completely.


> And why in this case a co_yield isn't needed?

This co_yield was formerly ```proceed::yes```, and we only need to yield ```proceed::yes``` when we're in the process of reading a variable or we're in a non_consuming state (you can see in ```continuous_data_consumer::process()``` that otherwise we're just returning to the coroutine after some checks), and here we're in neither